sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -295,36 +295,53 @@ def test_binomial():
     assert binomial(S.Half, S.Half) == 1
     assert binomial(-10, 1) == -10
     assert binomial(-10, 7) == -11440
-    assert binomial(n, -1).func == binomial
+    assert binomial(n, -1) == 0
     assert binomial(kp, -1) == 0
     assert binomial(nz, 0) == 1
     assert expand_func(binomial(n, 1)) == n
     assert expand_func(binomial(n, 2)) == n*(n - 1)/2
     assert expand_func(binomial(n, n - 2)) == n*(n - 1)/2
     assert expand_func(binomial(n, n - 1)) == n
-    assert binomial(n, 3).func == binomial
+    assert binomial(n, 3) == n*(n - 2)*(n - 1)/6
     assert binomial(n, 3).expand(func=True) ==  n**3/6 - n**2/2 + n/3
     assert expand_func(binomial(n, 3)) ==  n*(n - 2)*(n - 1)/6
     assert binomial(n, n) == 1
     assert binomial(n, n + 1).func == binomial  # e.g. (-1, 0) == 1
     assert binomial(kp, kp + 1) == 0
-    assert binomial(n, u).func == binomial
-    assert binomial(kp, u) == 0
-    assert binomial(n, p).func == binomial
+    assert binomial(n, u) == gamma(n + 1)/(gamma(u + 1)*gamma(n - u + 1))
+    assert binomial(kp, u) == gamma(kp + 1)/(gamma(u + 1)*gamma(kp - u + 1))
```

Comment:
I am not sure for this output, I suspect it to be `binomial(kp, u)`, since `u` can be a negative integer. Same for the above case.